test/functional/api/buffer_spec.lua

```diff
@@ -54,6 +60,15 @@ describe('buffer_* functions', function()
       eq({'a', 'b'}, curbuf('get_line_slice', 0, -1, true, false))
       eq({'b'}, curbuf('get_line_slice', 1, -1, true, false))
       eq({'b', 'c'}, curbuf('get_line_slice', -2, -1, true, true))
+      eq({}, curbuf('get_line_slice', 2, 3, false, true))
```

Comment:
@justinmk
Re: your first question. This is, indeed, how python works as well:
```python
lines = ['a', 'b', 'c']
lines[1:99] == ['b', 'c']
```
This behavior is also present in several other dynamic languages like Perl, JavaScript, and Ruby, so it's not without precedent. While, the compiled languages that I tried (Go, Java/Scala, Haskell, Octave, and Julia), all throw errors when slicing using an out-of-bounds index, it's not possible to mimic this behavior without transactions.

Re: `include_(start|end)`, I'm in favor of removing them since it's not clear what "start" and "end" mean. For instance, with the lines `[a, b, c]`, should 
`get_line_slice(1, 99, true, false)` return `[b, c]` or just `[b]`?
I think that having the client explicitly make these decisions would improve clarity.